index.js

```diff
@@ -88,9 +138,10 @@ CaseSensitivePathsPlugin.prototype.fileExistsWithCase = function(
 
   // If we are at the root, or have found a path we already know is good, return.
   if (
-    parsedPath.dir === parsedPath.root ||
-    dir === '.' ||
-    that.pathCache.has(filepath)
+    parsedPath.dir === parsedPath.root
+    || dir === '.'
+    // TODO: alternative way to express "known good path"
+    // || that.pathCache.has(filepath)
```

Comment:
@Urthen I'm not very sure if removing this line will cause any (noticeable) impact on performance.

`pathCache` is now efficiently used to prevent redundant calls to `fs.readdir()`. Forgetting(not memorying) known good paths causes more checks.

On my machine, the overall result is positive on performance.

However, if this becomes an issue we could do further work to memorize known good path.

How do you think about this?

Once we get a resolution I will change this line to be more readable.

Reply:
As far as I recall the `pathCache` needed to be primed with the current directory, and we needed to assume anything in the cache was correct. There was an issue where on windows you could CD into the directory with an incorrect case. While windows would allow it and otherwise there were no problems, when we were checking the case all the way down to root we would detect the incorrect case and fail the build unexpectedly. 

We'll need to keep the behavior of priming the cache & assuming the CWD is correctly cased in order to prevent a regression.

Reply:
That's reasonable, I will try to keep the `primeCache()` logic and use Promise on `pathCache` to achieve it.